kroxylicious-filters/kroxylicious-encryption/src/main/java/io/kroxylicious/filter/encryption/ResilientKms.java

```diff
@@ -80,7 +87,14 @@ private <A> CompletionStage<A> retry(String name, Supplier<CompletionStage<A>> o
         }
         Duration delay = strategy.getDelay(attempt);
         return schedule(operation, delay)
-                .exceptionallyComposeAsync(e -> retry(name, operation, attempt + 1));
+                .exceptionallyComposeAsync(e -> {
+                    if (e instanceof UnknownAliasException || e instanceof UnknownKeyException) {
+                        LOGGER.debug("not retrying unknown entity exception");
```

Comment:
We should really log which entity is unknown otherwise the logging is of no help when enabled.